packages/shared/src/components/notifications/Toast.tsx

```diff
@@ -47,11 +48,19 @@ const Toast = ({
         }
 
         await nextTick(); // wait (1ms) for the component to render so animation can be seen
-        startAnimation(data.timer);
       },
     },
   );
 
+  if (!testRef.current && toast?.message) {
+    testRef.current = toast;
+    startAnimation(toast.timer);
+  } else if (testRef.current && testRef.current !== toast && toast?.message) {
+    endAnimation();
+    testRef.current = toast;
+    startAnimation(toast.timer);
+  }
```

Comment:
```suggestion
  if(toast?.message) {
   if(testRef.current && testRef.current !== toast && toast?.message) {
   endAnimation();
   }
   
    testRef.current = toast;
    startAnimation(toast.timer);
  }
```

Reply:
From testing I can't see a way to refactor this as without the !testRef.current being top level it create a loop that keeps re doing the startAnimation